src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_upd_routines.F90

```diff
@@ -4403,6 +4465,217 @@ subroutine cat_enkf_increments(                               &
+                      ! (based on StieglitzSnow: htsn = (cpw*tsnow - fice*MAPL_ALHF)*swe )
+
+                      tmp_htsn(kk,n_e,isnow) = (0.0 - MAPL_ALHF)*tmp_wesn(kk,n_e,isnow)
+
+                      ! assign snow depth consistent with density of freshly fallen snow (must have SCF_ANA_MAXINCRSWE<=WEMIN)
+
+                      tmp_sndz(kk,n_e,isnow) = (WEMIN / RHOFS) / N_snow                   
+                      
+                   else
+                      
+                      ! snow in forecast and analysis, derive properties of analysis snow from properties of forecast snow
+                      
+                      ! update SWE:
+
+                      tmp_wesn(kk,n_e,isnow) = cat_progn(kk,n_e)%wesn(isnow) * swe_ratio
+
+                      ! update snow heat content:
+                      
+                      ! reichle, 27 Sep 2023: not sure if we can/should update htsn using swe_ratio;                       
+                      ! what if fcst snowpack contains liquid water?  
+                      ! in this case, would it make more sense to keep the temperature constant? 
+                      ! that is, do something like:
+                      !
+                      !   call StieglitzSnow_calc_tpsnow(1, cat_progn(kk,n_e)%htsn(isnow), cat_progn(kk,n_e)%wesn(isnow), snow_temp, fice)
+                      !
+                      !   tmp_htsn(kk, n_e, isnow) = (cpw*snow_temp - fice*MAPL_ALHF)*tmp_wesn(kk,n_e,isnow)
+                      !
+                      ! but not sure if the latter equation is correct 
+                      !
+                      ! for now, keep multiplication with swe_ratio:
+                      
+                      tmp_htsn(kk,n_e,isnow) = cat_progn(kk,n_e)%htsn(isnow) * swe_ratio  
+                      
+                      ! update snow depth: 
+
+                      if (asnow_ana < 1. .and. asnow_fcst < 1.) then
+                         
+                         ! keep snow depth constant when less than full snow cover in fcst and ana
+                         
+                         tmp_sndz(kk,n_e,isnow) = cat_progn(kk,n_e)%sndz(isnow)  
+                         
+                      else
+                         
+                         ! compute analysis snow depth by keeping snow density constant
+                         !
+                         ! in this case, it is possible that either asnow_fcst<1 or asnow_ana<1;
+                         ! when computing density or depth, make sure that SWE value (which is per unit area) is 
+                         ! adjusted to reflect SWE value (per unit area) in the snow-covered fraction of the tile
+                         
+                         ! i) diagnose (layer-specific) forecast snow density 
+                         
+                         snow_dens = ( cat_progn(kk,n_e)%wesn(isnow)/asnow_fcst ) / cat_progn(kk,n_e)%sndz(isnow) 
```

Comment:
This calculation now accounts for the possibility that asnow_fcst<1, which I think could be the case here.